buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -776,9 +778,8 @@ class BuildPlugin implements Plugin<Project> {
         }
     }
 
-    /** Returns a closure of common configuration shared by unit and integration tests. */
-    static Closure commonTestConfig(Project project) {
-        return {
+    static void applyCommonTestConfigPresentAndFuture(Project project) {
```

Comment:
We have other methods here which apply to all existing and future items of a type. I don't think we need such a verbose name for these cases. `configureTests` should be sufficient?

Reply:
In retrospect that does seem excessive. 
I think there is some value in coming up with something other than `apply` and `configure` to distinguish between what configures the current state vs attaching configuration to the domain object containers that affect future state.
For me this is one of the top confusions and sources of bugs, but we can do this in a follow up, I simplified the name for now.